Marlin/src/module/planner.cpp

```diff
@@ -792,17 +790,15 @@ block_t* Planner::get_current_block() {
  */
 void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t entry_speed, const_float_t exit_speed) {
 
-  const float spmm = block->steps_per_mm;
-  uint32_t initial_rate = entry_speed ? LROUND(entry_speed * spmm) : block->initial_rate,
-           final_rate = LROUND(exit_speed * spmm);
+  uint32_t initial_rate = LROUND(block->nominal_rate * entry_speed), // (steps per second)
+             final_rate = LROUND(block->nominal_rate * exit_speed);
 
   // Removing code to constrain values produces judder in direction-switching moves because the
   // current discrete stepping math diverges from physical motion under constant acceleration
   // when acceleration_steps_per_s2 is large compared to initial/final_rate.
```

Comment:
This comment probably should have been removed in PR #27089. That PR makes sure `min_entry_speed_sqr` is always respected which means the judder can't happen.

(before PR #27089 the `MINIMAL_STEP_RATE` of 120, which is fixed here, acted like a kludge/mitigation to prevent that judder)